src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/OptionsTests.cs

```diff
@@ -516,6 +516,60 @@ public static void MaxDepthRead()
             Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<BasicCompany>(BasicCompany.s_data, options));
         }
 
+        [Theory]
+        [InlineData(new char[] { '\n' }, '\r')]
+        [InlineData(new char[] { '\r', '\n' }, null)]
+        public static void NewLine(char[] newLineChars, char? notExpectedNewLineChars)
+        {
+            var obj = new BasicCompany();
+            obj.Initialize();
+
+            var newLine = new string(newLineChars);
+            var options = new JsonSerializerOptions();
+            var json = JsonSerializer.Serialize(obj, options);
+            Assert.DoesNotContain(newLine, json);
+
+            // Set custom newLine with indentation enabled
+            options = new JsonSerializerOptions();
+            options.WriteIndented = true;
+            options.NewLine = newLine;
+            json = JsonSerializer.Serialize(obj, options);
+            Assert.Contains(newLine, json);
+
+            if (notExpectedNewLineChars is { } notExpected)
+            {
+                Assert.DoesNotContain(notExpected, json);
+            }
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData(" ")]
+        [InlineData("\r")]
+        [InlineData("\n\n")]
+        [InlineData("\r\n\r\n")]
+        [InlineData("0")]
+        [InlineData("a")]
+        [InlineData("foo")]
+        [InlineData("$")]
+        [InlineData(".")]
+        [InlineData("\u03b1")]
+        public static void TestNewLineInvalid(string value)
+        {
+            var options = new JsonSerializerOptions();
+            Assert.Throws<ArgumentOutOfRangeException>("value", () => options.NewLine = value);
+        }
+
+        [Fact]
+        public static void TestNewLineNullResetsToDefault()
+        {
+            var options = new JsonSerializerOptions();
+            options.NewLine = "\n";
+            options.NewLine = null!;
```

Comment:
I find that behavior surprising. Assuming we want to use `null` to denote default that should be reflected in the nullability annotation for the property setter.

Reply:
Yeah I thought it a bit weird too, but annotations or not there was nothing to stop a user passing in null anyway.

Either the properties need changing to `string?`, or I can revert back to what I had originally and force the user to specify `Environment.NewLine` if they want to go back to the default.

Reply:
General practice is to throw argument null exception if an input parameter is non-nullable. I think using `Environment.NewLine` as the default works and results in a simpler API contract.

Reply:
I've updated things to do the normal "throw if null` behaviour, so the user would need to explicitly set to `Environment.NewLine` to restore the default behaviour.

Reply:
Thanks. We can always move to allow nulls in the future without introducing breaking changes.